app.js

```diff
@@ -1,55 +1,49 @@
-const computerChoiceDisplay = document.getElementById("computer-choice")
-const userChoiceDisplay = document.getElementById("user-choice")
-const resultDisplay = document.getElementById("result")
-const possibleChoices = document.querySelectorAll("button")
-let userChoice
-let computerChoice
-let result
+const resultMap = {
+    'rock': {
+        'paper': false,
+        'scissors': true,
+    },
+    'paper': {
+        'rock': true,
+        'scissors': false,
+    },
+    'scissors': {
+        'rock': false,
+        'paper': true
+    }
+};
 
-possibleChoices.forEach(possibleChoice => possibleChoice.addEventListener("click", (e) => {
-    userChoice = e.target.id
-    userChoiceDisplay.innerHTML = userChoice
-    generateComputerChoice()
-    getResult()
-}))
+const generateComputerChoice = () => {
+    const randomNumber = Math.floor(Math.random() * possibleChoices.length) + 1;
 
-function generateComputerChoice() {
-    const randomNumber = Math.floor(Math.random() * possibleChoices.length) + 1
+    switch (randomNumber) {
+        case 1:
+            return 'rock';
+        case 2:
+            return 'scissors';
+        case 3:
+            return 'paper';
+    };
+}
 
-    if (randomNumber === 1) {
-        computerChoice = "rock"
+const getResult = (choiceUser, choiceComputer) => {
+    if (choiceUser === choiceComputer) {
+        return "It's a draw!";
     }
-    if (randomNumber === 2) {
-        computerChoice = "scissors"
-    }
-    if (randomNumber === 3) {
-        computerChoice = "paper"
-    }
-    computerChoiceDisplay.innerHTML = computerChoice
+    const userWins = (resultMap[choiceUser][choiceComputer]);
```

Comment:
I'm extracting out the bool from the result map - the values represent whether the user wins or not.